tests/examples/common.py

```diff
@@ -0,0 +1,99 @@
+import os
+import tempfile
+import subprocess
+import sys
+
+# Get the path to the example binaries from an environment variable
+try:
+    MOCOCRW_EXAMPLE_BINARY_PATH = os.environ["MOCOCRW_EXAMPLE_BINARY_PATH"]
+except KeyError:
+    print("Please set the MOCOCRW_EXAMPLE_BINARY_PATH environment variable.", file=sys.stderr)
+    sys.exit(1)
+
+# Get the path to the example source dir from an environment variable
+try:
+    MOCOCRW_SRC_DIR = os.environ["MOCOCRW_SRC_DIR"]
+except KeyError:
+    print("Please set the MOCOCRW_SRC_DIR environment variable", file=sys.stderr)
+    sys.exit(1)
+
+
+HASH_ALGOS = [
+    "SHA256", "SHA384", "SHA512", "SHA3-256", "SHA3-384", "SHA3-512"
+]
+
+
+def add_invalid_parameters_and_execute(parameters):
+    parameters.extend(["--awöeljf", "asdölvkjyxc"])
+    result = subprocess.run(parameters, capture_output=True)
+    assert result.returncode == 1
+    assert b"Unknown option selected: unrecognised option '--aw\xc3\xb6eljf'\n" \
+           b"Failure parsing command line.\n" == result.stderr
+
+
+def create_temp_working_dir():
+    # Create working directory
+    new_working_dir = tempfile.mkdtemp()
+
+    current_working_dir = os.getcwd()
+    # Change to working directory
+    os.chdir(new_working_dir)
+    return current_working_dir, new_working_dir
+
+
+def combine_all_options(all_options, option_list):
+    """
+    This function combines the strings in the first string list with the elements of all other
+    string lists in all_options and adds the result to all_options.
+    e.g.
+    all_options = ""
+    option_list = [["hello"],["world"]
```

Comment:
Did you mean:
```suggestion
    option_list = [["hello"],["world"]]
```

Also, this function doesn't work for me as described:

```
In [1]: def combine_all_options(all_options, option_list):
   ...:     first_options_list = option_list[0]
   ...:     add_options = [add_opt + opt for add_opt in first_options_list for opt in all_options]
   ...:     all_options += add_options
   ...:     if len(option_list) == 1:
   ...:         return all_options
   ...:     return combine_all_options(all_options, option_list[1:])
   ...:

In [2]: combine_all_options("", [["hello"], ["world"]])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-aa0b89e2c631> in <module>()
----> 1 combine_all_options("", [["hello"], ["world"]])

<ipython-input-1-b76d49b2a128> in combine_all_options(all_options, option_list)
      2     first_options_list = option_list[0]
      3     add_options = [add_opt + opt for add_opt in first_options_list for opt in all_options]
----> 4     all_options += add_options
      5     if len(option_list) == 1:
      6         return all_options

TypeError: can only concatenate str (not "list") to str
```

I'm assuming you wanted `all_options` to be `[""]` initially, but even then, your function does not behave as you described it:

```
In [5]: combine_all_options([""], [["hello", "world"]])
Out[5]: ['', 'hello', 'world']
```

Can we not get rid of this entire function somehow? I'd really like to spend my time on reviewing MoCOCrW example code, not code that tests a command line interface for MoCOCrW crypto functionality.